### DISCUSSION TITLE
Custom blade directives / if statements

### DISCUSSION THREAD

**Discussion by maartenpaauw** [spatie/laravel-package-tools#33] [Ideas]
At the moment I'm creating a Laravel package that registers custom blade if statements. Which brought me to the idea to make it possible to register custom blade directives fluently via the package tool.

Maybe something like this (not sure about the method name yet):

```php
$package
    ->hasDirective('disk', function ($value) {
        return config('filesystems.default') === $value;
    })
```

The service provider will register the custom if statement via `Blade::if(...)`.

Let me know what you think. If I have some spare time I can create a PR.

https://laravel.com/docs/8.x/blade#custom-if-statements

**Comment by freekmurze** [comment]
Thanks for this suggestion.

This package tries to registering things very easily. I think the default way of registering these Blade components is easy enough for most, so I won't likely add it to this package.
